src/component.jl

```diff
@@ -868,3 +294,69 @@ function assign_new_uuid!(component::InfrastructureSystemsComponent)
     set_uuid!(get_internal(component), new_uuid)
     return
 end
+
+"""
+Attaches an info to a component
+"""
+function attach_supplemental_attribute!(
+    component::InfrastructureSystemsComponent,
+    info::T,
+) where {T <: InfrastructureSystemsSupplementalAttribute}
+    component_uuid = get_uuid(component)
+
+    if component_uuid ∈ info.components_uuid
+        throw(
+            ArgumentError(
+                "SupplementalAttribute type $T with UUID $(get_uuid(info)) already attached to component $(get_name(component))",
+            ),
+        )
+    end
+
+    push!(info.components_uuid, component_uuid)
+    infos_container = get_supplemental_attributes_container(component)
+
+    if !haskey(infos_container, T)
+        infos_container[T] = Set{T}()
+    end
+    push!(infos_container[T], info)
+    @debug "SupplementalAttribute type $T with UUID $(get_uuid(info)) stored in component $(get_name(component))"
+    return
+end
+
+"""
+Return true if the component has infos.
+"""
+function has_supplemental_attributes(component::InfrastructureSystemsComponent)
+    container = get_supplemental_attributes_container(component)
+    return !isempty(container)
+end
+
+function clear_supplemental_attributes!(component::InfrastructureSystemsComponent)
+    container = get_supplemental_attributes_container(component)
+    for info_set in values(container)
+        for i in info_set
+            delete!(get_components_uuid(i), get_uuid(component))
+        end
+    end
+    empty!(container)
+    @debug "Cleared infos in $(get_name(component))."
+    return
+end
+
+function remove_supplemental_attribute!(
+    component::InfrastructureSystemsComponent,
+    info::T,
+) where {T <: InfrastructureSystemsSupplementalAttribute}
+    container = get_supplemental_attributes_container(component)
+    if !haskey(container, T)
+        throw(
+            ArgumentError("info type $T is not stored in component $(get_name(component))"),
+        )
+    end
+    delete!(get_components_uuid(info), get_uuid(component))
+    delete!(container[T], info)
```

Comment:
Check if the attribute is present first?

Reply:
fixed via 0d561e6ff4ae61316208fdea2cf5732fd29ed419